### PR TITLE
LPD-2001 - c-prefers-expanded-text utility breaks styling in page editor

### PR DESCRIPTION
**Pull request by ilzamcmed**
Jira ticket: https://liferay.atlassian.net/browse/LPD-2001

https://github.com/user-attachments/assets/83dad867-bdf2-4ba1-9350-8ffbf228185f

### CODE REVIEW

**Comment by ilzamcmed** [comment]
ci:test:relevant

**Comment by ilzamcmed** [comment]
Hello everyone, @pat270 submitted [this PR](https://github.com/liferay/clay/pull/5622) to address the problem; however, there was no unanimous agreement on the solution. As a result, I've made the fix on the DXP side to resolve this specific issue.

**Comment by ilzamcmed** [comment]
ci:test:relevant

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/ItemConfiguration.js [hunk 1]
```diff
@@ -191,7 +212,12 @@ function ItemConfigurationContent({
 								key={panel.panelId}
 							>
 								<span
-									className="c-inner page-editor__page-structure__item-configuration-tab text-truncate"
+									className={classNames(
+										'c-inner page-editor__page-structure__item-configuration-tab',
+										{
+											'text-truncate': !isExpandedText,
```
Comment (pat270) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/ItemConfiguration.js:218]: @ilzamcmed we can't use `text-truncate` due to accessibility recommendation by @marcoscv-work. The full text should always display.

**Review by pat270 — CHANGES_REQUESTED** [review]

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/ItemConfiguration.js (hunk 1 above)
Comment (ilzamcmed) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/ItemConfiguration.js:218]: Oh, so it's a simpler change than I was expecting.
Thanks @pat270!

**Comment by ilzamcmed** [comment]
ci:test:sf

**Comment by ilzamcmed** [comment]
I am getting some errors while running `gw formatSource`. Investigating...

**Comment by ilzamcmed** [comment]
ci:test:sf

**Comment by ilzamcmed** [comment]
ci:test:sf

**Comment by ilzamcmed** [comment]
ci:test:sf

**Comment by ilzamcmed** [comment]
ci:test:relevant

**Comment by ilzamcmed** [comment]
I am investigating playwright errors.

**Comment by ilzamcmed** [comment]
ci:test:sf

**Comment by ilzamcmed** [comment]
ci:test:relevant

**Comment by ilzamcmed** [comment]
Hey @pat270 and @ethib137 
This PR is ready for review, again :smile: 
Thanks!

**Comment by ethib137** [comment]
Thanks @ilzamcmed . In this case, I don't think we want to take this approach though. We shouldn't need a one off fix in a single component for this. Whatever our fix is it should be more global and ideally take place in Clay. I think the only fix we may need would be removing `max-width: 100% !important;` from c-prefers-expanded-text. 

@pat270 is this really needed. In some basic testing it looks to me like it shouldn't be necessary.

Screenshot 2024-08-21 at 2 15 16 PM

**Comment by ilzamcmed** [comment]
Thanks @ethib137! I thought this solution to be more of a specific change since the one we should fix on clay didn't have any definition with @marcoscv-work on the other PR.
But you are correct! I Agree with a more global fix. 

Removing the `max-width: 100% !important;` makes the text break into a new line. Is that ok?

image

**Comment by pat270** [comment]
@ethib137 We can get rid of max-width: 100%. The reason why it's there is because `truncate-text` changes the display property to `block`, originally it is `inline`. `c-prefers-expanded-text` doesn't change it back to `inline`. If we overwrite to inline, we can remove the 100%. I'm not sure why I did it that way. I can't seem to find the edge case.

This one gets tricky because of the `flex-nowrap` customization on `nav-tabs`. We need to remove that. The tabs will break to new line when expanded like this:

will-break-to-new-line

**Comment by ethib137** [comment]
This is looking closer to the right solution. Our tabs on multiple lines does not look good. That probably needs to be rethought from a design perspective. 

@pat270 what's the problem with just removing `max-width: 100%` and not changing the display? 

Also, it looks like cadmin is overriding c-prefers-expanded-text.

In terms of removing c-inner from dxp, that's probably a good idea to create a ticket for it and add it to the backlog.

**Comment by ethib137** [comment]
Hey @pat270 your gif looks like you're adding `display: inline`. I'm suggesting just remove `max-width` nothing else.


https://github.com/user-attachments/assets/6dd61ec0-8c0d-46b7-a5fb-980c48191f72

**Comment by pat270** [comment]
@ethib137 You're right, we can remove it. It doesn't impact anything. Are you ok with the text just breaking to new line?

**Comment by ethib137** [comment]
Yeah, I think text breaking to new line is fine. In most cases that's what you want. At some point we can work on a tabs improvement to fix that, but it's not high priority.

**Comment by ilzamcmed** [comment]
To summarize:

1. Clay side: remove max-width: 100%
2. DXP side: 
    - undo the changes made on this PR 
    - create a task to remove `c-inner` class.

**Comment by pat270** [comment]
@ilzamcmed we don't need to remove `c-inner`. It will work fine with what Evan suggested. I can create a ticket for removing `c-inner` as a whole.

**Comment by ethib137** [comment]
Okay.. so we should be able to close this PR then right?

**Comment by ilzamcmed** [comment]
Thanks guys! :tada:

**Pull request closed without merging**